api/types/swarm/runtime/plugin.proto

```diff
@@ -0,0 +1,34 @@
+syntax = "proto3";
+
+import "gogoproto/gogo.proto";
+
+// GenericPluginPayload is used to wrap the PluginSpec with extra options before
+// scheduling on the cluster.
+// Its primary use is to pass along registry auth credentials from the client.
+message GenericPluginPayload {
+	PluginSpec spec = 1 [(gogoproto.nullable) = false];
+	PluginPullOptions PullOptions = 2 [(gogoproto.nullable) = false];
```

Comment:
Sorry if this is obvious, but why not have registry_auth in PluginSpec? I understand that it's not related to the plugin itself, it's just a way of getting the plugin, but one could argue that, it's the same for `remote`. I could view the PluginSpec as the struct that I need to get an actual plugin scheduled.